std/container/fixed_array.d

```diff
@@ -0,0 +1,504 @@
+module std.container.fixed_array;
+
+import core.exception, std.algorithm, std.conv, std.exception, std.range,
+        std.traits, std.typecons;
+
+public import std.container.util;
+version(unittest) import std.stdio, std.container;
+
+/**
+Array type that uses a $(XREF2 range, isRandomAccessRange, random access range)
+as fixed storage location. No memory is ever allocated. Throws if not enough
+space is left for an operation to succeed.
+*/
+struct FixedArray(Store)
+    if(isRandomAccessRange!Store && hasLength!Store)
+{
+
+/**
+    $(XREF2 range, isRandomAccessRange, random access range) that provides
+    the same range primitives as $(D Store).
+
+    $(LREF opIndex) and $(LREF opSlice) return values of this type.
+    */
+    alias Range = Store;
+
+/**
+ * The type of the elements that can be stored in this
+ * FixedArray.
+ */
+    alias Element = ElementType!Range;
+
+    private {
```

Comment:
We use `{` on its own line. With that the code becomes even taller, so I suggest either using the label form `private:` or prefixing each declaration with `private`.